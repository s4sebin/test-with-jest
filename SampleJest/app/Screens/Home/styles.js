import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',

    backgroundColor: 'white',
  },
  safeareaview: {
    flex: 1,
    backgroundColor: 'white',
  },

  buttonContainer: {
    padding: 15,
    margin: 10,
    borderRadius: 25,
    // width: '80%',
    alignSelf: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'grey',
  },
  title: {
    fontSize: 16,
  },
});
