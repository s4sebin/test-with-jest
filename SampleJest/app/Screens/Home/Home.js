import React, {useEffect, useState} from 'react';
import {Alert, Pressable, SafeAreaView, Text, TouchableOpacity, View} from 'react-native';
import styles from './styles.js';

export default function Home({navigation}) {
  return (
    <SafeAreaView style={styles.safeareaview}>
    <View style={styles.container}>
    <Pressable style={[styles.buttonContainer]} onPress={() => {
      navigation.navigate('Calculator');
    }}>
      <Text style={styles.title}>{"CALCULATOR"}</Text>
    </Pressable>
    </View>
    </SafeAreaView>
  );
}
