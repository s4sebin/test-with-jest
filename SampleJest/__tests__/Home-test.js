/**
 * @format
 */

import 'react-native';
import React from 'react';
import Home from '../app/Screens/Home/Home';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Home/>).toJSON();
  expect(tree).toMatchSnapshot();
});
