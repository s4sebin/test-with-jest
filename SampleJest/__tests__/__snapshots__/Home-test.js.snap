// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "grey",
            "borderRadius": 25,
            "justifyContent": "center",
            "margin": 10,
            "padding": 15,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        CALCULATOR
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
